File: 백준/Bronze/25377. 빵/빵.cc
#include <iostream>
using namespace std;
int main(void) {
    int n, a, b, time = 1001;
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> a >> b;
        if (a <= b && b < time)
            time = b;
    }
    cout << (time == 1001 ? -1 : time);
    return 0;
}

File: 백준/Bronze/25377. 빵/빵.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n=int(input())
time=1001
for _ in range(n):
    a,b=map(int,input().split())
    if a<=b and b<time:
        time=b
print(-1 if time==1001 else time)